### PR TITLE
SystemOrigin trait

### PR DESCRIPTION
**Pull request by pepyakin**
Those are 2 commits by Gav are cherry-picked from the [rococo-branch](https://github.com/paritytech/substrate/compare/270a992bf299cd41b2ed4e211814d2a958e3bfc8...rococo-branch).

Those are required to continue landing XCM v0 (See https://github.com/paritytech/polkadot/issues/1783)

### CODE REVIEW

File: frame/support/src/traits.rs [hunk 1]
```diff
@@ -1613,6 +1613,19 @@ pub trait UnfilteredDispatchable {
...
+	/// The type of account identifiers used by the system.
+	type AccountId;
+
+	/// Create with system none origin and `frame-system::Trait::BaseCallFilter`.
+	fn none() -> Self;
+	/// Create with system root origin and no filter.
+	fn root() -> Self;
+	/// Create with system signed origin and `frame-system::Trait::BaseCallFilter`.
+	fn signed(by: Self::AccountId) -> Self;
+}
```
Comment (shawntabrizi) [inline frame/support/src/traits.rs:1627]: Somehow this should be connected to this right?

https://github.com/paritytech/substrate/blob/master/frame/system/src/lib.rs#L320

Comment (pepyakin) [reply frame/support/src/traits.rs:1627]: Yup. I imagine that this connecting link is the macro that impls `SystemOrigin` above

**Review by shawntabrizi — APPROVED** [review]

**Review by thiolliere — COMMENTED** [review]
otherwise code looks good

**Review by thiolliere — APPROVED** [review]

File: frame/support/src/traits.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pepyakin) [reply frame/support/src/traits.rs:1627]: TBH, I am not aware why that distinction is made. Maybe @gavofyork can shed some more light on this?

[the remark this answers is not in the document]
Comment (shawntabrizi) [reply frame/support/src/traits.rs]: @thiolliere you are right and I updated

**Comment by pepyakin** [comment]
bot merge

**Pull request merged**